monai/bundle/workflows.py

```diff
@@ -268,6 +293,13 @@ def __init__(
                 f"Cannot find the metadata config file: {meta_file}. "
                 "Please see: https://docs.monai.io/en/stable/mb_specification.html"
             )
+        elif isinstance(meta_file, list):
```

Comment:
Should we use `self.meta_file` instead? To do you would move `meta_file=str(self.config_root_path / "metadata.json") if meta_file is None else meta_file` to be just before the `super()` call and then pass `meta_file=meta_file` as a new argument. I think the check for files existing here wouldn't be needed in this case.

Reply:
Hi @ericspod . Thanks for the suggestions. As for `isfile`, I've already updated the code.
However, it seems we cannot do `meta_file=str(self.config_root_path / "metadata.json") if meta_file is None else meta_file` before `super()`.
In order to do that, we also have to produce `self.config_root_path` first, but it would lead to an error in `__getattr__`.
What do you think?

Reply:
The whole `if config_file is not None:...` block could possibly be moved before the `super()` as well. It would set a local variable `config_root_path` (instead of `self.config_root_path`) which the meta file path can be constructed from. Then after `super()` have `self.config_root_path = config_root_path`.  This should work but may be getting messy now...

Reply:
Looks good to me now!